File: BookLocal.Data/Models/Service.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BookLocal.Data.Models
{
    public class Service
    {
        [Key]
        public int ServiceId { get; set; }

        [Required]
        public int BusinessId { get; set; }
        [ForeignKey("BusinessId")]
        public virtual Business Business { get; set; }

        [Required]
        public int ServiceCategoryId { get; set; }
        [ForeignKey("ServiceCategoryId")]
        public virtual ServiceCategory ServiceCategory { get; set; }

        [Required]
        [MaxLength(255)]
        public string Name { get; set; }
        public string? Description { get; set; }
        public bool IsArchived { get; set; } = false;

        public virtual ICollection<ServiceVariant> Variants { get; set; } = new List<ServiceVariant>();
    }
}

File: booklocal_mobile/lib/core/models/service.dart
import 'service_variant.dart';

class Service {
  final int id;
  final String name;
  final String? description;
  final int serviceCategoryId;
  final int businessId;
  final bool isArchived;
  final List<ServiceVariant> variants;

  Service({
    required this.id,
    required this.name,
    this.description,
    required this.serviceCategoryId,
    required this.businessId,
    required this.isArchived,
    required this.variants,
  });

  factory Service.fromJson(Map<String, dynamic> json) {
    return Service(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      description: json['description'],
      serviceCategoryId: json['serviceCategoryId'] ?? 0,
      businessId: json['businessId'] ?? 0,
      isArchived: json['isArchived'] ?? false,
      variants: (json['variants'] as List<dynamic>?)
              ?.map((v) => ServiceVariant.fromJson(v))
              .toList() ??
          [],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'serviceCategoryId': serviceCategoryId,
      'businessId': businessId,
      'isArchived': isArchived,
      'variants': variants.map((v) => v.toJson()).toList(),
    };
  }
}
